### PR TITLE
[Scan to update inventory] Part 2 – Product update

### PR DESCRIPTION
**Pull request by samiuelson**
#### [Scan to update inventory] Part 2 – Product's quantity update, unit tests & edge cases
This is **Part 2** of the Scan to update inventory project
Part 1 – https://github.com/woocommerce/woocommerce-android/pull/10250
Part 3 – https://github.com/woocommerce/woocommerce-android/pull/10278
Part 4 – https://github.com/woocommerce/woocommerce-android/pull/10313

### Description

Designs: w3JBReW1UJesCf02sj43kR-fi-775%3A34738

Changes:
1. Product stock quantity update
2. Showing error snackbars:
- when product is not stock-managed
- when product is not found for scanned sku
- when there was error during update operation
3. Unit tests for `FetchProductBySKU` and `ScanToUpdateInventoryViewModel`

### Testing instructions

1. Open product list
2. Click "barcode scanner" icon
3. Scan barcode with the SKU of existing product (you can use online barcode generator like [this one](https://barcode.tec-it.com/en/?data=ACGADO))
4. Verify product is found and the bottom sheet is opened
5. Click Quantity + 1 and verify the product quantity is updated successfully

In case of scanning the SKU of a product that is not stock-managed there should be a snack bar information shown.

### Images/gif

https://github.com/woocommerce/woocommerce-android/assets/4527432/2e8801c1-5b7f-410b-814f-d4a0d4af4ce7

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/inventory/ScanToUpdateInventoryViewModel.kt [hunk 1]
```diff
@@ -63,6 +71,20 @@ class ScanToUpdateInventoryViewModel @Inject constructor(
         }
     }
 
+    private suspend fun handleProductIsNotStockManaged(product: Product) {
+        triggerProductNotStockManagedSnackBar(product)
+        delay(SCANNER_RESTART_DEBOUNCE_MS)
```
Comment (rossanafmenezes) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/inventory/ScanToUpdateInventoryViewModel.kt:98]: 🔍 As far as I understand, the reason why we are restarting the scanner after the `ProductNotStockManagedSnackBar` is because the user would move to scan a different product, right? So you don't have to restart the scanner manually again by clicking on the icon. 

Maybe we could increase the delay by 2 or 3000? The way it's setup right now it's causing a weird state if you don't move the scanner away super fast from the barcode an you see the two snackbars popping out very fast. I had issues reading the error message in the beginning. 

https://github.com/woocommerce/woocommerce-android/assets/30724184/6541a28b-9011-410f-97d0-3b9186e9334e

WDYT?

**Review by rossanafmenezes — APPROVED** [review]
LGTM 

I left a small possible improvement regarding restarting the scanner. Let me know what you think.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/inventory/ScanToUpdateInventoryViewModel.kt (hunk 1 above)
Comment (samiuelson) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/inventory/ScanToUpdateInventoryViewModel.kt:98]: Great idea! I'm updating the debounce value to 3500L in another PR — https://github.com/woocommerce/woocommerce-android/pull/10278/commits/6513fc1031513c987a19ce15a01808b9c8e17f00
